### PR TITLE
WIP: Render first 32 measures of Bach Minuet

### PR DESCRIPTION
**Pull request by 0xfe**
This is a test to flesh out the `System`, `EasyScore` and `Factory` APIs. I hope to put the entire first part of the piece here. This will also be useful as a regression test when its done.

Current progress:

Open issues:
* Lots of duplication in building systems. `System` should support `measures` parameter so creating multi-measure systems is easy.
* Time signature is not vertically centered.
* Registry and IDs are super helpful. Adding classes to `EasyScore` keyvals will be enormously helpful configuring multiple similar notes.
* `EasyScore` keyvals for Articulations and Annotations may be helpful.
* ~~`StaveConnector` types need `Factory` shortcuts (like position and justification)~~
* Factory should register elements created by it instead of default registry.

### CODE REVIEW

**Comment by 0xfe** [comment]
I think this is in a reasonable state to merge (mainly because of the improvements to the other classes.)

I can continue to fix the open issues in later PRs.

File: tests/bach_tests.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      system.addStave({
+        voices: [
+          voice(notes('(G3 B3 D4)/h, A3/q', {clef: 'bass'}), {time: '3/4'})
+        ]
+      }).addClef('bass').addKeySignature('G').addTimeSignature('3/4');
+      system.addConnector().setType('brace');
+      system.addConnector().setType('single_right');
+      system.addConnector().setType('single_left');
+
+      id('d1').addModifier(0, vf.Annotation({text: 'p', vJustify: 'below'}));
```
Comment (Silverwolf90) [inline tests/bach_tests.js]: Real dynamics glyphs are available through `TextDynamics` or `TextNote`.

File: src/staveconnector.js [hunk 2]
```diff
@@ -49,6 +49,21 @@ export class StaveConnector extends Element {
     };
   }
 
+  static get typeString() {
+    return {
+      single_right: StaveConnector.type.SINGLE_RIGHT,
```
Comment (Silverwolf90) [inline src/staveconnector.js]: Can we move to camelCased aliases?

File: src/factory.js [hunk 3]
```diff
@@ -152,6 +156,51 @@ export class Factory {
     return acc;
   }
 
+  Annotation(params) {
+    params = setDefaults(params, {
+      text: 'p',
+      vJustify: 'below',
+      hJustify: 'center',
+      font_family: 'Times',
+      font_size: 14,
```
Comment (Silverwolf90) [inline src/factory.js]: Can we camelCase these too

**Comment by Silverwolf90** [comment]
What are you using for reference? Some stuff on IMSLP shows that the first bar has two voices in the left hand:

image

File: tests/bach_tests.js (hunk 1 above)
Comment (0xfe) [inline tests/bach_tests.js]: Done.

File: src/factory.js (hunk 3 above)
Comment (0xfe) [inline src/factory.js]: Done.

File: src/staveconnector.js (hunk 2 above)
Comment (0xfe) [inline src/staveconnector.js]: Done.

**Comment by 0xfe** [comment]
Reference: I have a Shirmer's Library copy of Bach pieces on my desk. :-)

**Comment by 0xfe** [comment]
Addressed your comments, PTAL.

**Comment by Silverwolf90** [comment]
Looks good to me!

**Comment by mscuthbert** [comment]
Great!

Resident musicologist here: the piece is not by Bach -- it's from a collection of pieces by other composers that JS Bach copied out into a notebook for his wife.  :-)

**Comment by 0xfe** [comment]
JSB: The original music pirate.

**Pull request merged**